src/module-elasticsuite-core/Index/Mapping/Field.php

```diff
@@ -320,21 +320,20 @@ private function getMultiFieldMappingPropertyConfig($analyzers)
      *
      * @return array
      */
-    private function getFieldAnalyzers()
+    private function getFieldAnalyzers(): array
     {
         $analyzers = [];
 
         if ($this->isSearchable() || $this->isUsedForSortBy()) {
             // Default search analyzer.
             $analyzers = [$this->getDefaultSearchAnalyzer()];
-
-            if ($this->isSearchable() && $this->getSearchWeight() > 1) {
-                $analyzers[] = self::ANALYZER_WHITESPACE;
-                $analyzers[] = self::ANALYZER_SHINGLE;
-            }
+        }
+        if ($this->isSearchable() && $this->getSearchWeight() > 1) {
+            $analyzers[] = self::ANALYZER_WHITESPACE;
+            $analyzers[] = self::ANALYZER_SHINGLE;
         }
 
-        if ($this->isFilterable() || empty($analyzers)) {
+        if (empty($analyzers) || $this->isFilterable()) {
```

Comment:
Why inverting the condition ?

Reply:
Condition "empty($analyzers)" execution costs less condition "$this->isFilterable()"